### ISSUE TITLE
Tab selection on Japanese IME on Mac

### ISSUE THREAD

**Issue by DaddyF** [standardnotes/forum#926]
When I use Japanese IME (Google), tab character is the selection key from the list of available candidates. But on Mac (latest Catalina), it works as a tab and the cursor moves while I am selecting candidates. On Windows10, it works fine. Please investigate.
Macbook Air (Early 2014), Catalina 10.15.4, English keyboard, standard notes 3.3.5
Google IME

**Comment by JaspalSuri** [comment]
I apologize if I'm misunderstanding but does this _only_ affect Standard Notes on macOS Catalina? I just want to make sure that this isn't an issue with Google IME. 

I noticed that there isn't a macOS app for it. Are you using the [Chrome extension](https://www.google.com/inputtools/chrome/)?

Can you provide a video or a list of steps for what you're trying to do? Thanks in advance! :)

**Comment by DaddyF** [comment]
Hi Jaspal,

I am using Windows 10 (bootcamp) and Catalina on Mac. At least I don’t see any issues on Windows 10.
On Catalina, I have screen recorded attached. So you may understand what I meant.

There’s another issue. Catalina has text expansion capability. I use it very often. This doesn’t work on Standard Notes, either.
I also use Apple’s Notes and I have no issues on it.
If you want more info, please let me know.

Sadao Fukatsu
[Screen Recording 2020-04-08 at 7.46.11.mov.zip](https://github.com/standardnotes/forum/files/4447355/Screen.Recording.2020-04-08.at.7.46.11.mov.zip)

**Comment by JaspalSuri** [comment]
Hi,

I'm not sure if you meant to close the issue on purpose. Thank you so much for the screen recording, I finally understand what you were trying to say earlier. 

My guess is that since Standard Notes is a non-native, cross-platform app that it doesn't support text expansion and similar third-party software. I'll check with the team to make sure that this is true.

**Comment by DaddyF** [comment]
I'm sorry that I accidentally closed this. I am not familiar with this yet. I still want your team to fix this issue along with the text expansion issue.
Thank you

**Comment by JaspalSuri** [comment]
Thanks for letting me know! I'll be sure to pass this feedback along to them!

**Comment by JaspalSuri** [comment]
I apologize for not including this in my first message. It looks like there isn’t a way to download Google Input Tools for macOS through any official channels. 

Fortunately, you can press `Command + Tab` and type `Keyboard` so that section of the System Preferences app will appear and should provide five tabs. 

Click on `Input Sources` > click on the `+` button (in the lower right hand corner) > search for the desired language (such as Japanese, in this case) > apply whatever options you'd find useful, then switch to them as needed in the menu bar of macOS. You should see something like the following: 

![System Preferences - Keyboard - Input Sources](https://user-images.githubusercontent.com/44656000/78739943-3d1a7780-790a-11ea-976d-74360e82af78.png)

I apologize if you've done this before but weren't satisfied with the results. This works within Standard Notes for me. I'll still ask the team about the text expansion issue! 

![Menubar Language Script Settings](https://user-images.githubusercontent.com/44656000/78739765-d1380f00-7909-11ea-9b57-e25e6e39615e.png)

**Comment by DaddyF** [comment]
Thank you very much for the information. It indeed works fine with Apple input method. I also found out that the text substitution works in Japanese mode. But this should work in English mode. I will continue to use this for a while, since this works fine.

**Comment by JaspalSuri** [comment]
That's great to hear! Thanks for letting us know!

> But this should work in English mode.
I completely understand and will give you an update as soon as I can (unless someone beats me to it).

**Comment by DaddyF** [comment]
I have further tested the Standard Notes and found a workaround. When you register a word in Japanese, then it works. But when you register a word in English, it has a limit of length of substitution. Manual says replacement word can have unto 64 characters. But it actually works more than that.  Please test it. I have attached screen recording for you to see.
[Screen Recording 2020-04-09 at 7.03.32.mov.zip](https://github.com/standardnotes/forum/files/4453047/Screen.Recording.2020-04-09.at.7.03.32.mov.zip)

**Comment by JaspalSuri** [comment]
Hello,

Are you saying that setting custom text substitutions in Japanese will work but it won't work in English? Is that what you've tried in the video or were both of those long texts provided by Apple? 

The text replacement limit appears to be a set limitation of macOS as their manual seems to be an exception to the rule. I would suggest providing them with [feedback](https://www.apple.com/feedback/) directly.

Thanks for providing that workaround! I'll bring this up with the team so they can take a closer look.

**Comment by DaddyF** [comment]
You are correct. English word substitution does not work, but Japanese word substitution works.
The substitution text is the one I picked from the web and pasted it. You may try it too, so you will fully understand. This test was done with Apple's Japanese input IME. However, Google IME is supported on OSX platform too. I have used it until you have suggested Apple's IME. It's a pity if it doesn't work on the Standard Notes. So something should be done for that. Otherwise, you might lose some users.

It was also interesting to test Microsoft word, Outlook how they behave for the apple's text substitution for comparison against Apple's apps.

**Comment by JaspalSuri** [comment]
Thank you for verifying! That makes sense and explains the reference to batteries. 

Do you have a link to Google's IME for macOS? 

> It's a pity if it doesn't work on the Standard Notes. So something should be done for that. Otherwise, you might lose some users.

I completely understand. I think this might be a limitation of either Electron (what's used to create the desktop apps for Slack, Standard Notes, Discord, etc.) or macOS. Given that the text substitution does not work in browsers such as Firefox, it may be worth looking into 3rd party programs that offer text expansion features such as PhraseExpress, Rocket Typist, or TextExpander. Please note that they are not endorsed nor affiliated with Standard Notes and please review their Privacy policies before using if you decide they're right for you.

**Comment by DaddyF** [comment]
Please refer to the URL below for Google IME for MacOS. It is in Japanese, but should be pretty obvious for the installation.
https://www.google.co.jp/ime/

I have also investigated text substitution apps for both Windows and MacOS. I use one on Windows. The beauty of the Standard Notes is that I can synch the data between them.

**Comment by JaspalSuri** [comment]
> Please refer to the URL below for Google IME for MacOS. It is in Japanese, but should be pretty obvious for the installation.
> 
> https://www.google.co.jp/ime/

Thank you so much for the link! I’ll be sure to test it out.

> I have also investigated text substitution apps for both Windows and MacOS. I use one on Windows. The beauty of the Standard Notes is that I can sync the data between them.

That’s true! I’ll try to look for something that works on both platforms and get back to you.

**Comment by JaspalSuri** [comment]
It looks like due to technical limitations or a bug, text expansion currently does not work in Standard Notes. 

To sync phrases between Windows and Mac, you can use PhraseExpress, according to [this page](https://www.phraseexpress.com/doc/share/cloud/). As a friendly reminder, Standard Notes is not affiliated with PhraseExpress nor endorses it; please read their privacy policy and terms of use before utilizing it. :)

**Comment by DaddyF** [comment]
Thank you for the feedback and the information about PhraseExpress. I am fine at this moment.

**Comment by JaspalSuri** [comment]
You’re welcome! Thank you for letting me know. Have a great weekend and stay safe!
